executor.go

```diff
@@ -275,6 +294,130 @@ func (e *Executor) provideEventLogger(t *Task) *Logger {
+	taskFlagsMap := make(map[string]FlagArg)
+
+	for _, flag := range flags {
+		var key string
+		var val string
+		splitFlag := strings.Split(flag, "=")
+
+		key, err := e.getVerifiedFlagKey(taskName, splitFlag[0])
+		// If the len(splitFlag) is 1, assume that it is a bool flag
+		if len(splitFlag) > 2 {
+			// If the passed flag has >1 "=" in it, this is invalid syntax
+			panic(fmt.Sprintf("Invalid flag syntax for %s: `%s`", taskName, flag))
+		} else if len(splitFlag) == 2 {
+			// If the passed flag has an "=" in it, assume that variable flag was set
+			// (e.g. --var="val")
+			val = splitFlag[1]
+		}
+
+		if err != nil {
+			panic(fmt.Sprintf("Unsupported flag passed to %s: `%s`. See error: %s", taskName, key, err))
+		}
+
+		// At this point, we should have verified that this flag is supported in `getVerifiedFlagKey`
+		taskFlag := e.taskFlagsRegistry[taskName][key]
+
+		// If a variable flag arg was not passed and a default exists, use the default
+		if val == "" && taskFlag.Default != "" {
+			val = taskFlag.Default
+		}
+
+		flagValErrMsg := fmt.Sprintf("The value for this flag is `%s`. Please use `%s`", taskFlag.ValueType, flagTypeToGetter[taskFlag.ValueType])
+		flagArg := FlagArg{
+			// Return the raw string val passed
+			Value: val,
+			BoolVal: func() bool {
+				if taskFlag.ValueType != "bool" {
+					panic(flagValErrMsg)
+				}
+
+				// If a bool flag is passed, it should alawys have a true value
+				return true
+			},
+			IntVal: func() int {
+				if taskFlag.ValueType != "int" {
+					panic(flagValErrMsg)
+				}
+
+				// Support flags that are passed either like `--flag=1` or `--flag="1"`
+				strippedVal := stripWrappingQuotations(val)
+				num, err := strconv.Atoi(strippedVal)
+				if err != nil {
+					panic(fmt.Sprintf("Please pass an integer as the value. Err: %s", err))
+				}
+				return num
+			},
+			StringVal: func() string {
+				if taskFlag.ValueType != "string" {
+					panic(flagValErrMsg)
+				}
+				// Support flags that are passed either like `--flag=val` or `--flag="val"`
+				strippedVal := stripWrappingQuotations(val)
+				return strippedVal
+			},
+		}
+		taskFlagsMap[key] = flagArg
+
+		// Allow readers to index into flag map via either LongName or ShortName
+		// regardless of which arg was passed in if both names are available
+		if len(key) == 1 && key != taskFlag.LongName && taskFlag.LongName != "" {
+			// If only one char was passed through, check whether a LongName is available
+			// and also register it in the map
+			key = taskFlag.LongName
+			taskFlagsMap[key] = flagArg
+		} else if len(key) > 1 && key == taskFlag.LongName && taskFlag.ShortName != 0 {
+			// If >1 char was passed through, check whether a ShortName is available
+			// and also register it in the map
+			key = string(taskFlag.ShortName)
+			taskFlagsMap[key] = flagArg
+		}
+	}
+
+	return taskFlagsMap
+}
+
+func stripWrappingQuotations(str string) string {
+	strippedVal := str
```

Comment:
i guess you could use `strings.Trim(str, "\"")` here but people could then pass in "asdf""""" and it would still work lol

also, wonder if we care about `'` (single quotes). mb not

another way you could write it is
```
if strings.HasPrefix("\"") && strings.HasSuffix("\"") {
    strippedVal = str[1 : len(str)-1]
}

Reply:
Oo in fact I did do `strings.Trim(...)` at first, but decided against it in case they wanted to include `""`s in their args for whatever reason

Didn't realize we had a `HasPrefix` and `HasSuffix` though :0